internal/providers/github/github.go

```diff
@@ -65,6 +74,7 @@ func NewRestClient(
 	ctx context.Context,
 	config *minderv1.GitHubProviderConfig,
 	metrics telemetry.HttpClientMetrics,
+	tokenStatStore *tokenstatus.Store,
```

Comment:
reading the rest of this module sounds like `tokenstatus.Store` is now required. Is that intended? If yes, then should we error if the store is not provided?

Alternatively, I wonder if it would make sense to rather create a `WithTokenStatus` option and handle its absence in the REST code.

Reply:
> reading the rest of this module sounds like tokenstatus.Store is now required. Is that intended?'

Yes, it is. I'll add a null check.

> Alternatively, I wonder if it would make sense to rather create a WithTokenStatus option and handle its absence in the REST code.

Why would it be absent? The only case where I think it may be absent is when we don't want to handle rate limit errors. Is this something we should support? For me, if we don't want to be blocked, we can pass a context with a short timeout (in case we are blocked by rate limit)

Reply:
To be honest I was only wondering if we could decrease the number of places where we pass the `tokenstatus.Store` by using something like the `WithTokenStatus` option and then just have a passthrough implementation as a default opt to avoid increasing the number of required parameters across the codebase. There's no functional reason.